### PR TITLE
Extend log.file.rotation.date parser

### PR DESCRIPTION
**Pull request by michaelklishin**
## Proposed Changes

to successfully parse all cases presented in Lager docs.

## Types of Changes

- [x] Bug fix (non-breaking change which fixes issue #3831)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)
- [ ] Build system and/or CI

## Checklist

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass locally with my changes
- [ ] II have added necessary documentation to https://github.com/rabbitmq/rabbitmq-website
- [x] I have added this change to the first version(s) in release-notes that I expect to introduce it

## Further Comments

References #3831. Note that this is only the parser part, actual rotation might need still be adjusted.

### CODE REVIEW

**Comment by michaelklishin** [comment]
We do not handle hourly rotation as far as I can tell but it should be fine as a first step towards reasonable Lager rotation spec feature parity. Again, we recommend using `logrotate` and similar tools as the built-in rotation will never match them in terms of feature set and maturity.

File: deps/rabbit/apps/rabbitmq_prelaunch/test/rabbit_logger_std_h_SUITE.erl [hunk 1]
```diff
@@ -210,3 +207,58 @@ every_month_rotation_is_detected(_) ->
...
+          error,
+          rabbit_logger_std_h:parse_date_spec("$$D0")),
+      ?assertEqual(
+          error,
+          rabbit_logger_std_h:parse_date_spec("$D99")).
+
+parse_date_spec_case6(_) ->
+      ?assertEqual(
+          #{every => day, hour => 0, minute => 30},
+          rabbit_logger_std_h:parse_date_spec("$H30")),
```
Comment (lukebakken) [inline deps/rabbit/apps/rabbitmq_prelaunch/test/rabbit_logger_std_h_SUITE.erl:258]: Should this result in the following map? I'm using this as a reference - https://github.com/erlang-lager/lager#internal-log-rotation

```
#{every => hour, minute => 30}
```

File: deps/rabbit/apps/rabbitmq_prelaunch/test/rabbit_logger_std_h_SUITE.erl [hunk 2]
```diff
@@ -210,3 +207,58 @@ every_month_rotation_is_detected(_) ->
...
+      ?assertEqual(
+          #{every => day, hour => 0, minute => 3},
+          rabbit_logger_std_h:parse_date_spec("$H3")),
```
Comment (lukebakken) [inline deps/rabbit/apps/rabbitmq_prelaunch/test/rabbit_logger_std_h_SUITE.erl:261]:
```
#{every => hour, minute => 3}
```

**Review by lukebakken — COMMENTED** [review]
I have a couple questions about `Hmm` parsing.

**Comment by lukebakken** [comment]
👍 great!

**Pull request merged**